bundles/binding/org.openhab.binding.ddwrt/src/main/java/org/openhab/binding/ddwrt/internal/ddwrtBinding.java

```diff
@@ -0,0 +1,542 @@
+                    receive(client); // user:
+                    send(client, username);
+                }
+                receive(client); // password:
+                logger.trace("TelnetCommandThread password ({})", password);
+                send(client, password);
+                receive(client); // welcome text
+                send(client, cmdString);
+                Thread.sleep(1000L); // response not needed - may be interesting
+                                     // for reading status
+
+                // There is a ddwrt problem on restarting of virtual networks. So we have to restart the lan service.
+                if (type.startsWith("wlanguest") && !interface_guest.isEmpty() && command == OnOffType.ON) {
+                    cmdString = "stopservice lan && startservice lan";
+                    send(client, cmdString);
+                    Thread.sleep(25000L); // response not needed but time for restarting
+                }
+
+                logger.trace("TelnetCommandThread ok send");
+                client.disconnect();
+
+            } catch (
+
+            Exception e) {
+                logger.warn("Error processing command", e);
+            }
+        }
+
+        private void send(TelnetClient client, String data) {
+            logger.trace("Sending data ({})...", data);
+            try {
+                data += "\r\n";
+                client.getOutputStream().write(data.getBytes());
+                client.getOutputStream().flush();
+            } catch (IOException e) {
+                logger.warn("Error sending data", e);
+            }
+        }
+
+        private String receive(TelnetClient client) {
+            StringBuffer strBuffer;
+            try {
+                strBuffer = new StringBuffer();
+
+                byte[] buf = new byte[4096];
+                int len = 0;
+
+                Thread.sleep(750L);
+
+                while ((len = client.getInputStream().read(buf)) != 0) {
+                    strBuffer.append(new String(buf, 0, len));
+
+                    Thread.sleep(750L);
+
+                    if (client.getInputStream().available() == 0) {
+                        break;
+                    }
+                }
+
+                return strBuffer.toString();
+
+            } catch (Exception e) {
+                logger.warn("Error receiving data", e);
+            }
+
+            return null;
+        }
+    }
+
+    /**
+     * A quartz scheduler job to simply do a reconnection to the ddwrt.
+     */
+    public static class ReconnectJob implements Job {
+
+        @Override
+        public void execute(JobExecutionContext arg0) throws JobExecutionException {
+            INSTANCE.conditionalDeActivate();
+        }
+
+    }
+
+    @Override
+    protected void execute() {
+
+        logger.trace("execute");
+        if (password == null) {
+            return;
+        } else if (password.trim().isEmpty()) {
+            return;
+        }
```

Comment:
You could use `StringUtils.isNotBlank(password)` here, and perhaps a TRACE or DEBUG log explaining why the poll was aborted?